src/datatable.py

```diff
@@ -0,0 +1,192 @@
+                uwu = ""
+                if col == "_id":
+                    if self.group_by == "fullgnomen":
+                        uwu = '<a href="/variant/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "dn_no":
+                        uwu = '<a href="/patient/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "gene":
+                        uwu = '<a href="/gene/{0}">{0}</a>'.format(val)
+                elif col == "total":
+                    uwu = val
+                else:
+                    # filtering like this doesn't work yet
+                    # uwu = '<a href="/all?{1}={0}">{0}</a>'.format(val, col)
+                    uwu = val
+                data_row.append(uwu)
+            data_rows.append(data_row)
+
+        output['data'] = data_rows
+        return output
+
+    def run_queries(self):
+        pages = self.paging()  # pages has 'start' and 'length' attributes
+        filter = self.filtering()  # the term entered into the datatable search
+        sorting = self.sorting()  # the document field chosen to sort
+
+        # get result from db
+        match = {"$match": filter}
+        group = {}
+        if self.group_by is not None:
+            group = {
+                "$group": {
+                    "_id": "$" + self.group_by,
+                    "total": {
+                        "$sum": 1
+                    }
+                }
+            }
+
+        if self.group_by == "fullgnomen":
+            group["$group"]["protein"] = {"$addToSet": "$pnomen"}
+        pipeline = []
+
+        if group:
+            pipeline.append(group)
+        if filter:
+            pipeline.append(match)
+        pipeline = pipeline + [{"$sort": sorting}, {"$skip": pages.start}]
+        if pages.length >= 0:
+            pipeline.append({"$limit": pages.length})
+        self.result_data = list(self.dbh[self.collection].aggregate(pipeline))
+
+        if group:
+            # total amount unfiltered & total amount filtered (search bar)
+            self.records_total = len(list(self.dbh[self.collection].aggregate([group])))
+            self.records_filtered = len(list(self.dbh[self.collection].aggregate([group, match])))
+        else:
+            self.records_total = len(list(self.dbh[self.collection].find()))
+            self.records_filtered = len(list(self.dbh[self.collection].find(filter)))
+
+    def filtering(self):
+        """
+        build your filter spec
+        "search": {
+            "regex": "false",
+            "value": ""
+        },
+        :return: filter dict
+        """
+        # TODO : search per individual column? not needed?
+        filter = {}
+        if self.request['search']['value'] != "":
+            # need to match for every field
+            # the term put into search is logically concatenated with 'or' between all columns
+            or_filter_on_all_columns = []
+            for k, v in self.request["columns"].items():
```

Comment:
Are the variables `k` and `v`  short-lived variables or loop-index variables? Or, could the variable have a more meaningful name?

Reply:
see earlier comment:
I have made another issue #19, so that the next version needs to have a look.

Reply:
I also tried to explain this a bit more in an earlier comment.
`datatables.py` is a very generic script. Any datatable in the VUSualizer is made based on the information that needs to be shown at that time. It takes information (for example dictionaries), but you do not know beforehand what it is. are these genes? variants? patients? or something else you search for in the search bar?
I think that is why the developer choose to use variable names like "key" and "value", because it is a dictionary with unknown content that is being called.
key and value, or k and v, are accepted names for these dictionary variables within the python community.